sdk/iot/Azure.Iot.Hub.Service/src/DevicesClient.cs

```diff
@@ -13,13 +17,489 @@ namespace Azure.Iot.Hub.Service
+            String ifMatchHeaderValue = IfMatchPreconditionExtensions.GetIfMatchHeaderValue(precondition, twinUpdate.Etag);
+            return _twinClient.UpdateDeviceTwin(twinUpdate.DeviceId, twinUpdate, ifMatchHeaderValue, cancellationToken);
+        }
+
+        /// <summary>
+        /// Update multiple devices' twins. A maximum of 100 updates can be done per call, and each operation must be done on a different device twin. For larger scale operations, consider using IoT Hub jobs (https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry#import-and-export-device-identities).
+        /// </summary>
+        /// <param name="twinUpdates">The new twins to replace the twins on existing devices.</param>
+        /// <param name="precondition">The condition for which each operation will execute.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The result of the bulk operation.</returns>
+        public virtual async Task<Response<BulkRegistryOperationResult>> UpdateTwinsAsync(IEnumerable<TwinData> twinUpdates, BulkIfMatchPrecondition precondition = BulkIfMatchPrecondition.IfMatch, CancellationToken cancellationToken = default)
+        {
+            IEnumerable<ExportImportDevice> registryOperations = twinUpdates
+                .ToList()
+                .Select(x => new ExportImportDevice()
+                {
+                    Id = x.DeviceId,
+                    Tags = x.Tags,
+                    Properties = new PropertyContainer(x.Properties?.Desired, x.Properties?.Reported),
+                    TwinETag = x.Etag,
+                    ImportMode = precondition == BulkIfMatchPrecondition.Unconditional ? ExportImportDeviceImportMode.UpdateTwin : ExportImportDeviceImportMode.UpdateTwinIfMatchETag
+                });
+
+            return await _registryManagerClient.BulkDeviceCrudAsync(registryOperations, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Update multiple devices' twins. A maximum of 100 updates can be done per call, and each operation must be done on a different device twin. For larger scale operations, consider using IoT Hub jobs (https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-identity-registry#import-and-export-device-identities).
+        /// </summary>
+        /// <param name="twinUpdates">The new twins to replace the twins on existing devices.</param>
+        /// <param name="precondition">The condition for which each operation will execute.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The result of the bulk operation.</returns>
+        public virtual Response<BulkRegistryOperationResult> UpdateTwins(IEnumerable<TwinData> twinUpdates, BulkIfMatchPrecondition precondition = BulkIfMatchPrecondition.IfMatch, CancellationToken cancellationToken = default)
```

Comment:
Should we change this type to be `BulkRegistryOperationResponse`? or even better, should we just remove the term `Registry` here? since it doesn't make sense anymore now that we have encapsulated the registry manager terminologies?

Reply:
For Result vs Response, sure I like Response, too.

Reply:
I like removing the 'Registry' but maybe just post on the teams channel if everyone is onboard or has any concerns.

Reply:
you would just change the type name in the custom code right?

Reply:
I could change it in the custom code, sure

Reply:
I'll add it to the modified swagger file as well, so we can hopefully not have custom code for this in the future